### ISSUE TITLE
IDropTarget.DragOver fires as expected, but Drop never fires

### ISSUE THREAD

**Issue by WillSullivan** [punker76/gong-wpf-dragdrop#185]
Two list boxes bound to two different collections.  One is the source, the other the target.  The target has ```AllowDrop="True"``` (I'm cargo culting at this point) and a DragDrop.DropHandler bound to a property on the view model.

(quick aside--the view model in this case is a wrapper around a model that allows me to attach properties used in the UI, such as a drop handler to a model.  Also, tons of fat trimmed from example code).

```xml
<ListBox AllowDrop="True" ItemsSource="{Binding HERPS}" dd:DragDrop.IsDropTarget="True" dd:DragDrop.DropHandler="{Binding DropHandler}" />
<ListBox ItemsSource="{Binding Configuration.DERPS}" dd:DragDrop.IsDragSource="True" />
```

The drop handler is simple

```c#
public sealed class DropHandler : IDropTarget
{
	public const string DropHandlerPropertyName = "DropHandlerCommand";
	private readonly BaseModelViewModel _target;

	public DropHandler(BaseModelViewModel target)
	{
		_target = target;
	}
	public void DragOver(IDropInfo dropInfo)
	{
		dropInfo.DropTargetAdorner = DropTargetAdorners.Insert;
		dropInfo.Effects = DragDropEffects.Link;
	}

	public void Drop(IDropInfo dropInfo)
	{
		var handler = _target.GetPropertyValue(DropHandlerPropertyName) as ICommand;
		if (handler == null) return;
		handler.Execute(dropInfo);
	}
}
```
For each wrapped *something* I grab their drop handler (an ICommand implementation) on drop and pass in the drop info.  Nothing crazy here.

Now, when I drag an item from DERPS over into HERPS, ```IDropTarget.DragOver(IDropInfo dropInfo)``` does get called.  I can see that ```dropInfo.Data``` contains the correct data, and that ```dropInfo.TargetCollection``` is the correct collection.  Everything else looks correct as well, except for TargetItem and VisualTargetItem being null (not sure if that's bad or not).

I can also see a visual indication that an insert is going to happen within the list box.  There is a line that indicates where the item will be inserted.  So everything appears nice and happy, but....

The mouse cursor never changes from the "no drop" icon when hovering over the drop target.  And when I release the mouse, ```IDropTarget.Drop(IDropInfo dropInfo)``` does not fire.

Any guesses why?  I didn't see anything in the samples that stood out as "DO THIS OR IT DONT WORK".  Any hints as to what I can do to debug this further?  I'm a bit at a loss, and am about at the point where I have to strip everything out and re-add it bit by bit to see what breaks the drag/drop magic :(

**Comment by WillSullivan** [comment]
HAH!  Repro'd in a new project.  You can too.

Create a new wpf project, add the nuget package for the latest version of dragdrop.  Here's the main window xaml

```xml
<Window x:Class="DragAndDerp.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:DragAndDerp"
        xmlns:dd="clr-namespace:GongSolutions.Wpf.DragDrop;assembly=GongSolutions.Wpf.DragDrop"
        mc:Ignorable="d"
        Title="MainWindow" Height="350" Width="525">
    <Grid>
        <Grid.ColumnDefinitions>
            <ColumnDefinition />
            <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
            <RowDefinition/>
            <RowDefinition Height="auto"/>
        </Grid.RowDefinitions>
        <ListBox ItemsSource="{Binding BoxOne}" dd:DragDrop.IsDropTarget="True" dd:DragDrop.DropHandler="{Binding DropHandler}" />
        <ListBox Grid.Column="1" ItemsSource="{Binding BoxTwo}" dd:DragDrop.IsDragSource="True" />
        <TextBox Text="{Binding Status}" Grid.Row="1" Grid.ColumnSpan="2" />
    </Grid>
</Window>
```

Here's the codebehind with everything crammed in

```c#
namespace DragAndDerp
{
	/// <summary>
	/// Interaction logic for MainWindow.xaml
	/// </summary>
	public partial class MainWindow : Window
	{
		public MainWindow()
		{
			InitializeComponent();
			DataContext = new ViewModel();
		}
	}

	public class ViewModel : INotifyPropertyChanged
	{
		private string _status;
		public ObservableCollection<string> BoxOne { get; } = new ObservableCollection<string>();
		public ObservableCollection<string> BoxTwo { get; } = new ObservableCollection<string>();
		public ViewModel()
		{
			BoxTwo.Add("what");
			BoxTwo.Add("the");
			BoxTwo.Add("fudge");
			DropHandler = new DropTarget(this);
		}

		public string Status
		{
			get
			{
				return _status;
			}
			set
			{
				_status = value;
				PropertyChanged?.Invoke(this,new PropertyChangedEventArgs(nameof(Status)));
			}
		}

		public IDropTarget DropHandler { get; private set; }

		public event PropertyChangedEventHandler PropertyChanged;
	}

	public class DropTarget : IDropTarget
	{
		private readonly ViewModel _parent;

		public DropTarget(ViewModel parent)
		{
			_parent = parent;
		}
		public void DragOver(IDropInfo dropInfo)
		{
			_parent.Status = "over";
		}

		public void Drop(IDropInfo dropInfo)
		{
			_parent.Status = "dropped";
			_parent.BoxOne.Add(dropInfo.Data as string);
		}
	}
}
```

This all looks right to me, but it exhibits the exact same behavior I was talking about above.  What am I doing wrong?

**Comment by punker76** [comment]
@WillSullivan Do you tested the latest pre-release?

**Comment by WillSullivan** [comment]
Sorry, 1.4.3

**Comment by WillSullivan** [comment]
@punker76 Just cloned locally, added a reference, and it still is bugged.  Going to step through and see if it's something obvious/easy to fix.

**Comment by WillSullivan** [comment]
Sigh.  
```dropInfo.NotHandled = true;```

Closing.

**Comment by punker76** [comment]
@WillSullivan ?

**Comment by WillSullivan** [comment]
@punker76 I set ```dropInfo.NotHandled``` to True in the DragOver and it works.  Not sure if that is not supposed to be required in order for it to work.

**Comment by jstevenson72** [comment]
THANKS!  None of the examples appear to give this missing piece of critical data...

**Comment by WillSullivan** [comment]
@jstevenson72 hey, if you know of a good place to add this info, think about requesting a pull with it added.

**Comment by RedX2501** [comment]
I do realize that this is a very old topic but it fits my problem exactly.

I'm using 2.4.3

If I set in DragOver the `dropInfo.Effects = DragDropEffects.Link` it shows the drop not allowed symbol and never triggers `Drop` when I release the mouse over the element. Changing `Link` to `Move` immediately works.

It does not matter what I set `dropInfo.NotHandled` to.

What magic is needed for `Link` to work?
